src/librbd/internal.cc

```diff
@@ -1913,6 +1921,20 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force) {
       unknown_format = false;
       id = ictx->id;
 
+      if (force) {
+        C_SaferCond ctx;
+        ictx->exclusive_lock->shut_down(&ctx);
```

Comment:
The reason I'm shutting down `exclusive_lock` here is because it needs to be done before the call to `ictx->operations->prepare_image_update()`

Why shouldn't the `exclusive_lock` be initialized at this point? Since ImageState::open() is synchronous, I though it would not be a problem to do this here.